sdk/compiler/damlc/daml-ide/src/DA/Daml/LanguageServer/SplitGotoDefinition.hs

```diff
@@ -216,16 +219,15 @@ gotoDefinitionByName ideState params = do
     file <-
       hoistMaybe (Just "Failed to find module") $
         find (isSuffixOf moduleSuffix . ("/" <> ) . fromNormalizedFilePath) sortedSrcFiles
+    tc <- useOrThrow "Failed to typecheck" TypeCheck file
     
-    -- It might be better to get the typechecked module and look for the identifier in there?
-    spans <- useOrThrow "Failed to get span info" GetSpanInfo file
     let expectedOccName = mkOccName (fromTryGetDefinitionNameSpace $ tgdnIdentifierNameSpace $ gdnpName params) (tgdnIdentifierName $ gdnpName params)
-        locations = 
+        locations =
           [ srcSpanToLocation $ nameSrcSpan name
-          | Just name <- getNameM . spaninfoSource <$> spansExprs spans
-          , expectedOccName == nameOccName name && isGoodSrcSpan (nameSrcSpan name)
```

Comment:
I think this specific line change (226) is a noop?

Reply:
ah true, i had rewritten this from start, so the fact that it slightly lines up is coincidence. I don't think the order matters here and prefer `actual == expected`

Reply:
:+1: